admin/services/data-access/pupil-census-import.data.service.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const { TYPES } = require('tedious')
+const moment = require('moment')
+
+const config = require('../../config')
+const pupilCensusImportDataService = {}
+const sqlService = require('./sql.service')
+const sqlPoolService = require('./sql.pool.service')
+
+/**
+ * Execute pupil data bulk import
+ * @param {Array} pupilData
+ * @param {Array} schools
+ * @return {Promise<*>}
+ */
+pupilCensusImportDataService.sqlBulkImport = async(pupilData, schools) => {
+  const result = {}
+  const con = await sqlPoolService.getConnection()
+
+  try {
+    result.output = await bulkLoadData(con, pupilData, schools)
+  } catch (error) {
+    result.errorOutput = error
+  }
+  con.release()
+  return result
+}
+
+const bulkLoadData = (connection, pupilData, schools) => {
+  return new Promise((resolve, reject) => {
+    const bulkLoad = connection.newBulkLoad(`${config.Sql.Database}.${sqlService.adminSchema}.[pupil]`, function (error, rowCount) {
+      if (error) {
+        reject(error)
```

Comment:
If we reject here, we should also return... to prevent the subsequent resolve()

Reply:
@jon-shipley updated